tests/org.pitest.pitclipse.ui.tests/src/org/pitest/pitclipse/ui/swtbot/PitResultNotifier.java

```diff
@@ -16,36 +16,75 @@
 
 package org.pitest.pitclipse.ui.swtbot;
 
+import java.io.IOException;
+
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
+import org.eclipse.swtbot.swt.finder.utils.SWTBotPreferences;
+import org.eclipse.swtbot.swt.finder.waits.DefaultCondition;
 import org.pitest.pitclipse.core.extension.point.ResultNotifier;
-import org.pitest.pitclipse.ui.extension.point.PitUiUpdate;
+import org.pitest.pitclipse.runner.PitResults;
 import org.pitest.pitclipse.ui.swtbot.ResultsParser.Summary;
 
-public class PitResultNotifier implements ResultNotifier<PitUiUpdate> {
-    @Override
-    public void handleResults(PitUiUpdate updateEvent) {
-        notifiyTestsOfHtmlResults(updateEvent);
-    }
+/**
+ * @author Jonas Kutscha
+ */
+public class PitResultNotifier implements ResultNotifier<PitResults> {
+    public enum PitSummary {
+        INSTANCE;
 
-    private void notifiyTestsOfHtmlResults(PitUiUpdate updateEvent) {
-        PitResultsView view = buildResultsView(updateEvent);
-        tryNotifyResults(view);
-    }
+        private Summary summary;
+        private SWTWorkbenchBot bot;
+
+        public int getClasses() {
+            bot.waitUntil(new SummaryWaitCondition(), SWTBotPreferences.TIMEOUT);
```

Comment:
I'm not sure I like this new strategy: before we were waiting for the summary to be ready and then we were accessing the results. Now, we don't explicitly wait for the summary to be ready but we implicitly wait for it when retrieving classes AND coverage AND mutation coverage (of course, only the first wait will be a real wait, but still...)... since we must be aware of resetting, why shouldn't be aware also of the wait before accessing the results? Or am I missing something?

Reply:
This class is not responsible for waiting for pit to finish. These methods are only called at one place and prior to that we are waiting for pit to finish. These waits are only in place to wait for a short while to make up for the desynchronization between the extension point and our waiting mechanism.